File: server/manager/admin-js-sdk/form/SettingForm.js
'use strict';



/**

*/
class SettingForm {

	/**
	 * 广告（首页）
	 */
    notice:String;

	/**
	 * 广播（跑马灯）
	 */
    radio:String;

	/**
	 * 充值信息
	 */
    payInfo:String;

	/**
	 * 用户协议
	 */
    agreement:String;

	/**
	 * 用户默认房卡
	 */
    initGold:Number;
    constructor() {

    }

    formObject({notice,radio,payInfo,agreement,initGold}):SettingForm{
        this.notice = notice;
        this.radio = radio;
        this.payInfo = payInfo;
        this.agreement = agreement;
        this.initGold = initGold;
        return this;
    }

    static of({notice,radio,payInfo,agreement,initGold}):SettingForm{
        return new SettingForm().formObject({notice,radio,payInfo,agreement,initGold});
    }

    static form(notice:String,radio:String,payInfo:String,agreement:String,initGold:Number):SettingForm{
        return new SettingForm().formObject({notice,radio,payInfo,agreement,initGold});
    }
}

export default SettingForm;

File: server/project/mj-client/src/main/java/majiang/client/portal/admin/form/SettingForm.java
package majiang.client.portal.admin.form;

import org.forkjoin.apikit.core.Message;

import javax.validation.constraints.NotNull;

/**
 *
 */
@Message
public class SettingForm {
    /**广告（首页）*/
    @NotNull
    private String notice;

    /**广播（跑马灯）*/
    @NotNull
    private String radio;

    /**充值信息*/
    @NotNull
    private String payInfo;

    /**用户协议*/
    @NotNull
    private String agreement;

    /**用户默认房卡*/
    private int initGold;

    public String getNotice() {
        return notice;
    }

    public void setNotice(String notice) {
        this.notice = notice;
    }

    public String getRadio() {
        return radio;
    }

    public void setRadio(String radio) {
        this.radio = radio;
    }

    public String getPayInfo() {
        return payInfo;
    }

    public void setPayInfo(String payInfo) {
        this.payInfo = payInfo;
    }

    public String getAgreement() {
        return agreement;
    }

    public void setAgreement(String agreement) {
        this.agreement = agreement;
    }

    public int getInitGold() {
        return initGold;
    }

    public void setInitGold(int initGold) {
        this.initGold = initGold;
    }

    @Override
    public String toString() {
        return "SettingForm{" +
                "notice='" + notice + '\'' +
                ", radio='" + radio + '\'' +
                ", payInfo='" + payInfo + '\'' +
                ", agreement='" + agreement + '\'' +
                ", initGold=" + initGold +
                '}';
    }
}
